<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect><x>0</x><y>0</y><width>1200</width><height>700</height></rect>
  </property>
  <property name="windowTitle"><string>Veil IPS/IDS Dashboard</string></property>

  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="mainLayout">

    <!-- ðŸŸ¦ Sidebar -->
    <item>
     <widget class="QFrame" name="sidebarFrame">
      <property name="styleSheet">
       <string notr="true">
        QFrame {
         background-color: #1a1b26;
         color: #c0caf5;
        }
        QPushButton {
         background-color: transparent;
         color: #c0caf5;
         text-align: left;
         padding: 8px 20px;
         border: none;
         font-size: 14px;
         border-radius: 6px;
         <!-- transition: all 0.3s; -->
        }
        QPushButton:hover {
         background-color: #292e42;
        }
        QLabel {
         color: #7aa2f7;
         padding: 12px;
         font-weight: bold;
         font-size: 15px;
        }
       </string>
      </property>

      <layout class="QVBoxLayout" name="sidebarLayout">
       <item><widget class="QLabel" name="menuTitle"><property name="text"><string>VEIL Dashboard</string></property></widget></item>
       <item><widget class="QPushButton" name="btnDashboard"><property name="text"><string>Dashboard</string></property></widget></item>
       <item><widget class="QPushButton" name="btnNotifications"><property name="text"><string>Notifications</string></property></widget></item>
       <item><widget class="QPushButton" name="btnNetwork"><property name="text"><string>Network Traffic</string></property></widget></item>
       <item><widget class="QPushButton" name="btnControls"><property name="text"><string>Prevention &amp; Response</string></property></widget></item>
       <item><widget class="QPushButton" name="btnRules"><property name="text"><string>Rules Management</string></property></widget></item>
       <item><widget class="QPushButton" name="btnReports"><property name="text"><string>Report Analytics</string></property></widget></item>
       <item><spacer name="verticalSpacer"><property name="orientation"><enum>Qt::Vertical</enum></property></spacer></item>
       <item><widget class="QPushButton" name="btnSettings"><property name="text"><string>Settings</string></property></widget></item>
      </layout>
     </widget>
    </item>

    <!-- ðŸŸ© Main Content -->
    <item>
     <widget class="QFrame" name="mainContentFrame">
      <property name="styleSheet">
       <string notr="true">
        QFrame {
         background-color: #24283b;
         color: #c0caf5;
        }
        QLabel {
         color: #c0caf5;
         font-size: 14px;
        }
        QPushButton {
         background-color: #3d59a1;
         color: white;
         border: none;
         border-radius: 6px;
         padding: 6px 12px;
        }
        QPushButton:hover {
         background-color: #7aa2f7;
        }
       </string>
      </property>

      <layout class="QVBoxLayout" name="mainContentLayout">
       <item>
        <widget class="QStackedWidget" name="stackedWidget">

         <!-- ðŸŸ¨ Dashboard Page -->
         <widget class="QWidget" name="pageDashboard">
          <layout class="QVBoxLayout" name="dashboardLayout">

           <!-- Header -->
           <item>
            <widget class="QFrame" name="headerFrame">
             <property name="styleSheet">
              <string notr="true">
               QFrame {
                background-color: #1f2335;
                border-bottom: 1px solid #3b4261;
               }
               QLabel {
                font-size: 18px;
                font-weight: bold;
                color: #7aa2f7;
               }
              </string>
             </property>
             <layout class="QHBoxLayout" name="headerLayout">
              <item><widget class="QLabel" name="headerTitle"><property name="text"><string>Dashboard Overview</string></property></widget></item>
              <item><spacer name="headerSpacer"><property name="orientation"><enum>Qt::Horizontal</enum></property></spacer></item>
              <item><widget class="QPushButton" name="refreshButton"><property name="text"><string>Refresh</string></property></widget></item>
             </layout>
            </widget>
           </item>

           <!-- Glass Cards -->
           <item>
            <widget class="QFrame" name="statsFrame">
             <property name="styleSheet">
              <string notr="true">
               QFrame {
                background: transparent;
               }
               QLabel {
                color: #c0caf5;
                font-size: 13px;
               }
              </string>
             </property>
             <layout class="QHBoxLayout" name="statsLayout">

              <!-- Network -->
              <item>
               <widget class="QFrame" name="cardNetwork">
                <property name="styleSheet">
                 <string notr="true">
                  QFrame {
                   background-color: rgba(255,255,255,0.05);
                   border-radius: 10px;
                   padding: 20px;
                   border: 1px solid #3b4261;
                  }
                  QLabel { color: #c0caf5; font-size: 13px; }
                 </string>
                </property>
                <layout class="QVBoxLayout" name="networkCardLayout">
                 <item><widget class="QLabel" name="labelNetworkTitle"><property name="text"><string>Network Status</string></property></widget></item>
                 <item><widget class="QLabel" name="labelNetworkValue"><property name="text"><string>Stable</string></property></widget></item>
                </layout>
               </widget>
              </item>

              <!-- Traffic -->
              <item>
               <widget class="QFrame" name="cardTraffic">
                <property name="styleSheet">
                 <string notr="true">
                  QFrame {
                   background-color: rgba(255,255,255,0.05);
                   border-radius: 10px;
                   padding: 20px;
                   border: 1px solid #3b4261;
                  }
                 </string>
                </property>
                <layout class="QVBoxLayout" name="trafficCardLayout">
                 <item><widget class="QLabel" name="labelTrafficTitle"><property name="text"><string>Traffic Load</string></property></widget></item>
                 <item><widget class="QLabel" name="labelTrafficValue"><property name="text"><string>245 Mbps</string></property></widget></item>
                </layout>
               </widget>
              </item>

              <!-- Health -->
              <item>
               <widget class="QFrame" name="cardHealth">
                <property name="styleSheet">
                 <string notr="true">
                  QFrame {
                   background-color: rgba(255,255,255,0.05);
                   border-radius: 10px;
                   padding: 20px;
                   border: 1px solid #3b4261;
                  }
                 </string>
                </property>
                <layout class="QVBoxLayout" name="healthCardLayout">
                 <item><widget class="QLabel" name="labelHealthTitle"><property name="text"><string>System Health</string></property></widget></item>
                 <item><widget class="QLabel" name="labelHealthValue"><property name="text"><string>Good</string></property></widget></item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
           </item>

           <!-- Chart Section -->
           <item>
            <widget class="QFrame" name="chartFrame">
             <property name="styleSheet">
              <string notr="true">
               QFrame {
                background-color: #1f2335;
                border-radius: 10px;
                padding: 20px;
                border: 1px solid #3b4261;
               }
               QLabel { color: #9aa5ce; }
              </string>
             </property>
             <layout class="QVBoxLayout" name="chartLayout">
              <item><widget class="QLabel" name="chartTitle"><property name="text"><string>Network Activity (Last 24 Hours)</string></property></widget></item>
              <item><widget class="QWidget" name="graphWidget"><property name="minimumSize"><size><width>400</width><height>250</height></size></property></widget></item>
             </layout>
            </widget>
           </item>

          </layout>
         </widget>

         <!-- Other Pages -->
         <widget class="QWidget" name="pageNotifications"><layout class="QVBoxLayout" name="notifLayout"><item><widget class="QLabel" name="labelNotif"><property name="text"><string>Notifications content goes here.</string></property></widget></item></layout></widget>
         <widget class="QWidget" name="pageNetwork"><layout class="QVBoxLayout" name="networkLayout"><item><widget class="QLabel" name="labelNetwork"><property name="text"><string>Network Traffic Analysis content here.</string></property></widget></item></layout></widget>
         <widget class="QWidget" name="pageControls"><layout class="QVBoxLayout" name="controlsLayout"><item><widget class="QLabel" name="labelControls"><property name="text"><string>Prevention &amp; Response content here.</string></property></widget></item></layout></widget>
         <widget class="QWidget" name="pageRules"><layout class="QVBoxLayout" name="rulesLayout"><item><widget class="QLabel" name="labelRules"><property name="text"><string>Rules Management content here.</string></property></widget></item></layout></widget>
         <widget class="QWidget" name="pageReports"><layout class="QVBoxLayout" name="reportsLayout"><item><widget class="QLabel" name="labelReports"><property name="text"><string>Report Analytics content here.</string></property></widget></item></layout></widget>
         <widget class="QWidget" name="pageSettings"><layout class="QVBoxLayout" name="settingsLayout"><item><widget class="QLabel" name="labelSettings"><property name="text"><string>Settings content here.</string></property></widget></item></layout></widget>

        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>

  <widget class="QMenuBar" name="menubar"/>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
